java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1120,6 +1122,24 @@ private boolean hasLocallyPresentOptionalBinding(ResolvedBindings resolvedBindin
               && !getLocalExplicitBindings(keyFactory.unwrapOptional(resolvedBindings.key()).get())
                   .isEmpty();
         }
+
+        /**
+         * Returns {@code true} if {@code resolvedBindings} contains a provision binding
+         * for which there is an explicit present binding in this component that was resolved
+         * previously.
+         */
+        boolean hasLocallyConflictingContributionBinding(ResolvedBindings resolvedBindings) {
+          return resolvedBindings
+                .contributionBindings()
+                .stream()
+                .anyMatch(contributionBinding -> contributionBinding.bindingKind().equals(PROVISION)
+                  && !getLocalExplicitBindings(resolvedBindings.key()).isEmpty()
+                  && parentResolver.isPresent()
```

Comment:
But isn't that the point? If we do find one of those bindings, then there are *some* local bindings, and we should return `true`. We're not actually putting any of these bindings into the `resolvedBindings` map at this point anyway.

Or am I missing something?